experiments/obp_connector/09_obp_connector.ipynb

```diff
@@ -0,0 +1,658 @@
+      "/home/hdilab01/hdiRecSys/ReplayHDILab/replay/optuna_objective.py:76: FutureWarning: suggest_uniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/tag/v3.0.0. Use :func:`~optuna.trial.Trial.suggest_float` instead.\n",
+      "  res[param] = suggest_fn(param, low=low, high=high)\n",
+      "/home/hdilab01/anaconda3/envs/rec_sys/lib/python3.9/site-packages/pyspark/sql/pandas/conversion.py:289: UserWarning: createDataFrame attempted Arrow optimization because 'spark.sql.execution.arrow.pyspark.enabled' is set to true; however, failed by the reason below:\n",
+      "  Unsupported type in conversion from Arrow: uint8\n",
+      "Attempting non-optimization as 'spark.sql.execution.arrow.pyspark.fallback.enabled' is set to true.\n",
+      "  warnings.warn(msg)\n",
+      "\u001b[32m[I 2023-07-19 20:26:23,177]\u001b[0m Trial 28 finished with value: 0.005333333333333333 and parameters: {'eps': -4.352363633806379, 'alpha': 8.93924160537841}. Best is trial 1 with value: 0.005333333333333333.\u001b[0m\n",
+      "/home/hdilab01/hdiRecSys/ReplayHDILab/replay/optuna_objective.py:76: FutureWarning: suggest_uniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/tag/v3.0.0. Use :func:`~optuna.trial.Trial.suggest_float` instead.\n",
+      "  res[param] = suggest_fn(param, low=low, high=high)\n",
+      "/home/hdilab01/anaconda3/envs/rec_sys/lib/python3.9/site-packages/pyspark/sql/pandas/conversion.py:289: UserWarning: createDataFrame attempted Arrow optimization because 'spark.sql.execution.arrow.pyspark.enabled' is set to true; however, failed by the reason below:\n",
+      "  Unsupported type in conversion from Arrow: uint8\n",
+      "Attempting non-optimization as 'spark.sql.execution.arrow.pyspark.fallback.enabled' is set to true.\n",
+      "  warnings.warn(msg)\n",
+      "\u001b[32m[I 2023-07-19 20:26:47,883]\u001b[0m Trial 29 finished with value: 0.0008888888888888889 and parameters: {'eps': 0.2728175134206978, 'alpha': 9.84088973098143}. Best is trial 1 with value: 0.005333333333333333.\u001b[0m\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{'eps': -6.078942275169988, 'alpha': 0.1117365525776511}\n"
+     ]
+    }
+   ],
+   "source": [
+    "#Define borders for the optimized parameters\n",
+    "param_borders = {\n",
+    "                \"eps\": [-10, 10],\n",
+    "                \"alpha\": [0.001, 10]\n",
+    "}\n",
+    "\n",
+    "#Take subset of train data to validate our model with OBP\n",
+    "bandit_feedback_subset = bandit_subset([0, 300000], bandit_feedback_train) #The first parameter is a slice of subset [a, b]\n",
+    "print(learner.optimize(bandit_feedback_subset, val_size=0.3, param_borders=param_borders, budget=30))"
```

Comment:
When will you add in this PR LinUCB and other imported models?

1. They are needed for reproducible runs of notebook and scripts
2. Now optimize in notebook looks strange: it took 2nd trial out of 30 and resulting epsilon is negative.

Reply:
1) Yes, sure it is a mistake. At that moment, we should only use models which are currently available in RePlay;
2) Actually negative eps is ok for LinUCB algorithm because in offline setting bandit algorithms work better with pessimism rather than optimism.